src/ghost.js

```diff
@@ -0,0 +1,186 @@
+      // const { payload, issuer } = await this._verifyData(data)
+      // TODO: solve jwt verification problem
+      const { payload } = this.decode(data)
+      if (payload) {
+        switch (payload.type) {
+          case 'join':
+            this._userJoined(payload.iss, from)
+          break;
+          case 'request_backlog':
+            this.sendDirect(from, { type: 'backlog', message: this.backlog })
+          break;
+          default:
+            this._messageReceived(payload.iss, payload)
+        }
+      }
+    })
+    this._room.on('peer joined', (peer) => this.announce(peer))
+    this._room.on('peer left', (peer) => this._userLeft(peer))
+  }
+
+  /**
+   * Get name of the chat
+   *
+   * @return    {String}      chat name
+   */
+  get name () {
+    return this._name
+  }
+
+  /**
+   * Get name of the chat
+   *
+   * @return    {String}      chat name
+   */
+  get peerId () {
+    return this._peerId
+  }
+
+  /**
+   * Get all users online
+   *
+   * @return    {Array<String>}      users online
+   */
+  get onlineUsers () {
+    return Object.keys(this._usersOnline).filter(id => !id.startsWith('Qm'))
+  }
+
+  /**
+   *
+   *
+   * @return    {String}      ipfs peer id
+   */
+  threeIdToPeerId (did) {
+    return this._usersOnline[did]
+  }
+
+  /**
+   * Get backlog of all past messages
+   *
+   * @return    {Array<Object>}      users online
+   */
+  get backlog () {
+    return [...this._backlog]
+  }
+
+  // Announce entry in chat and share our 3id and peerID, empty jwt suffices
+  async announce (to) {
+    !to ? await this.broadcast({ type: 'join' })
+    : await this.sendDirect(to, { type: 'join' })
+  }
+
+  /**
+   * Post a message to the thread
+   *
+   * @param     {Object}    message                 The message
+   * @param     {String}    to                      PeerID to send the message to (optional)
+   * @return    {String}                            The postId of the new post
+   */
+  async post (message, to) {
+    !to ? await this.broadcast({ type: 'chat', message })
+    : await this.sendDirect(to, { type: 'chat', message })
+  }
+
+  /**
+   * Request a backlog of past messages from peers in the chat
+   *
+   */
+  async requestBacklog () {
+    await this.broadcast({ type: 'request_backlog' })
+  }
+
+  /**
+   * Leave the chat
+   *
+   */
+  async leaveChat () {
+    // await this.broadcast({ type: 'leave' })
+    await this._room.leave()
+  }
+
+  /**
+   * Broadcast a message to peers in the room
+   *
+   * @param     {Object}    message                 The message
+   */
+  async broadcast (message) {
+    const jwt = await this._3id.signJWT(message, { use3ID: true })
+    this._room.broadcast(jwt)
+  }
+
+  /**
+   * Send a direct message to a peer
+   *
+   * @param     {String}    peerID              The PeerID of the receiver
+   * @param     {Object}    message             The message
+   */
+  async sendDirect (to, message) {
```

Comment:
This should also be private I think. And since this method exists I don't think we need a `to` in the `post` method.

Reply:
I made `sendDirect` private but I do think think `post` should be kept how it is, it provides a public way to send both broadcasted and direct messages.